server/src/main/java/org/apache/druid/server/http/CoordinatorResource.java

```diff
@@ -111,14 +111,23 @@ public Response getLoadQueue(
       return Response.ok(
           Maps.transformValues(
               coordinator.getLoadManagementPeons(),
-              input -> {
-                long loadSize = input.getSizeOfSegmentsToLoad();
-                long dropSize = input.getSegmentsToDrop().stream().mapToLong(DataSegment::getSize).sum();
+              peon -> {
+                long loadSize = peon.getSizeOfSegmentsToLoad();
+                long dropSize = peon.getSegmentsToDrop().stream().mapToLong(DataSegment::getSize).sum();
+
+                // 1 kbps = 1/8 kB/s = 1/8 B/ms
+                long loadRateKbps = peon.getLoadRateKbps();
+                long expectedLoadTimeMillis
+                    = loadRateKbps > 0 && loadSize > 0
+                      ? (8 * loadSize) / loadRateKbps
+                      : 0;
+
                 return new ImmutableMap.Builder<>()
-                    .put("segmentsToLoad", input.getSegmentsToLoad().size())
-                    .put("segmentsToDrop", input.getSegmentsToDrop().size())
+                    .put("segmentsToLoad", peon.getSegmentsToLoad().size())
+                    .put("segmentsToDrop", peon.getSegmentsToDrop().size())
                     .put("segmentsToLoadSize", loadSize)
                     .put("segmentsToDropSize", dropSize)
+                    .put("expectedLoadTimeMillis", expectedLoadTimeMillis)
                     .build();
               }
```

Comment:
Curious why some of these stats are only available in the `simple` mode. I see with `full`, the response being returned currently is less flexible to add new things in this code...

Reply:
Actually, the `full` mode is currently broken. It just returns the entire `LoadQueuePeon` object in the response.
`CuratorLoadQueuePeon` had some `@JsonProperty` fields that were serialized out into the response.
But `HttpLoadQueuePeon` has never had any `@JsonProperty` fields since it was first written, so we just get an empty object in the response of the API `/druid/v1/coordinator/loadQueue?full`.

Since no one has ever reported this issue, I assume no one is using it (the web-console certainly doesn't use it).
I will create a separate PR to either fix it or just get rid of it completely.